src/backend/cpu/device_manager.cpp

```diff
@@ -129,4 +138,18 @@ DeviceManager& DeviceManager::getInstance() {
 
 CPUInfo DeviceManager::getCPUInfo() const { return cinfo; }
 
+void DeviceManager::setMemoryManager(
+    std::unique_ptr<MemoryManagerBase> newMgr) {
+    // Get the current memory manager to prevent deadlock if it hasn't been
+    // initialized yet
+    memoryManager();
```

Comment:
I am a bit confused here. 

`DeviceManager` is a singleton and we can only it's method once we have an object of `DeviceManager`. Thus, the very call that get's us the object of type `DeviceManager` instantiates the singleton and in turn the `memManager` member variable.

I don't think this call is needed here at all. Please let me know if I missed anything in the call hierarchy.

Reply:
I added this because of problems I had in my original implementation with how `af_init` interacted with the, `std::call_once` to generate the manager - there was some deadlock on Windows CPUf for whatever reason. I'll remove this for now and add it back if I run into the same issue.